ecolex/legislation.py

```diff
@@ -190,20 +178,32 @@ def harvest_file(upfile):
                 legislation['legCountry_es'] = fao_country.get('es')
                 legislation['legCountry_fr'] = fao_country.get('fr')
 
-        legYear = legislation.get('legYear')
-        if legYear:
+                region = json_regions.get(fao_country.get('en'))
+                if region:
+                    legislation['legGeoArea_en'] = region.get('en', [])
+                    legislation['legGeoArea_fr'] = region.get('fr', [])
+                    legislation['legGeoArea_es'] = region.get('es', [])
+
+        if 'legDate' in legislation:
             try:
-                legDate = datetime.strptime(legYear, '%Y')
+                legDate = datetime.strptime(legislation.get('legDate'), '%Y-%m-%d')
```

Comment:
@raressiteavu  @iuliachiriac This is not ok, there are a ton of records having just the year or YYYY-MM in Date_of_Text

<Date_of_Text>2012</Date_of_Text>
<Date_of_Text>2012-00-5</Date_of_Text>
<Date_of_Text>2012-07-02</Date_of_Text>
<Date_of_Text>2012-00-00</Date_of_Text>

We need to extract whatever useful information can be extracted from that field - at least the year.
2012-00-5